### ISSUE TITLE
\emoji does not support emoji with ZWJ

### ISSUE THREAD

**Issue by canton7** [doxygen/doxygen#8918]
**Describe the bug**

`\emoji :woman_firefighter:`, or some other emoji which consist of two emoji joined with a ZWJ, renders incorrectly in the HTML and XML output (I haven't checked others), as Doxygen misses out the ZWJ.

E.g. :woman_firefighter: is encoded as:

 - 👩 U+1F469
 - U+200D (the ZWJ)
 - 🚒 U+1F692

However, the HTML output misses out the ZWJ, so it just gets rendered as 👩🚒. The XML output also misses out the ZWJ: `<emoji name="woman_firefighter" unicode="&amp;#x1f469;&amp;#x1f692;"/>`.

**Expected behavior**

The emoji is rendered as 👩‍🚒 in HTML output, and as `<emoji name="woman_firefighter" unicode="&amp;#x1f469;&amp;#x200d;&amp;#x1f692;"/>` in the XML output.

**To Reproduce**

Just put `\emoji :woman_firefighter:` in any doxygen comment.

I'm happy to create a repro if required.

**Version**

1.9.2 (caa4e3de211fbbef2c3adf58a6bd4c86d0eb7cb8) Windows 10 64-bit. Installer, from the website.

**Comment by albert-github** [comment]
The problem looks related to the used emoji set. In the doxygen code I see with the list of supported emoji by doxygen: `/* generated using the download_github_emojis.py script */` though I cannot find the mentioned script. When looking at the, probable, source of the emoji (i.e. github) and some results:

woman firefighter:
```
- :woman_firefighter:
- &#x1f469;&#x1f692;
- &#x1f469;&zwj;&#x1f692;
```
results in (on github):
- :woman_firefighter:
- &#x1f469;&#x1f692;
- &#x1f469;&zwj;&#x1f692;

Though it is not a problem for all combined symbols e.g.:
Antarctica:
```
- :antarctica:
- &#x1f1e6;&#x1f1f6;
- &#x1f1e6;&zwj;&#x1f1f6;
```
results in (on github):
- :antarctica:
- &#x1f1e6;&#x1f1f6;
- &#x1f1e6;&zwj;&#x1f1f6;

and doxygen gives for:
```
\emoji :woman_firefighter:

\emoji :antarctica:
```

![image](https://user-images.githubusercontent.com/5223533/143682318-3684e8f4-a7d7-48bf-8812-246a5f3e546e.png)

So it looks like the used source has some problems. 

I think the original emoji implementation comes from #6517 and some rework on it.

@doxygen 
- where can I find the script: `download_github_emojis.py` so I can have a look at the current github codes?
- should we adjust the code of the different emoji that don't concatenate properly at the moment (in case of a name change the old, for compatibility, and the new name should be present)?
   - like woman_firefighter with a `&zwj;`
   - and the more complex basketball_woman (it will be a 5 position code, see e.g https://unicode.org/emoji/charts/full-emoji-list.html woman bouncing ball or https://emojipedia.org/woman-bouncing-ball/ )?

**Comment by doxygen** [comment]
@albert-github I've attached the script for reference
[download_github_emojis.zip](https://github.com/doxygen/doxygen/files/7620656/download_github_emojis.zip)

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).


@doxygen thanks for the script, so it is just the script that was in the documentation.

**Comment by canton7** [comment]
Thank you!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
